backend/src/apiserver/client_manager.go

```diff
@@ -120,7 +123,7 @@ func (c *ClientManager) init() {
 	c.runStore = storage.NewRunStore(db, c.time)
 	c.resourceReferenceStore = storage.NewResourceReferenceStore(db)
 	c.dBStatusStore = storage.NewDBStatusStore(db)
-	c.objectStore = initMinioClient(getDurationConfig(initConnectionTimeout))
```

Comment:
Hi,

MinioClient (https://docs.minio.io/docs/minio-client-complete-guide) provides an abstraction that can be used to make calls to: 
- S3
- GCS
- Minio Server (on-prem object store: https://www.minio.io/)
- Other object stores. 

Would you consider modifying the code so that all calls are made through Minio Client, and some configuration determines whether S3, GCS, or Minio Server is used? 

This is what, for instance, Argo has done to abstract away the object store (see https://github.com/argoproj/argo/blob/master/ARTIFACT_REPO.md). 

It would be a great contributions to abstract away the object store in the same way here. WDYT?

Reply:
This sounds reasonable to me. I tried to use `minio` client for accessing S3 objects but had an issue related to using EC2 instance profile authentication. Thanks for pointing me out to Argo repo, I think this will be useful.
I will explore the approach they use, in order to develop generic abstraction for different object storages.
For now, could we consider the proposed solution as temporal?

Reply:
Hi @DmitryBe,

We do want to move all the GCS calls to Minio Client. Would it be possible to undertake this as part of this PR? 

It will: 
- Reduce the amount of code
- Support out of the box all the cloud platforms (Azure, S3, etc.)
- Exercise most of the code through our e2e tests (only the cloud specific configuration won't be tested).

Reply:
ok, I will work on it.

Reply:
Thank you! This is great help.